packages/page-experience/lib/PageDataGatherer.js

```diff
@@ -0,0 +1,236 @@
+/**
+ * Copyright 2021  The AMP HTML Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const puppeteer = require('puppeteer');
+const treeKill = require('tree-kill');
+const parseFontfaces = require('./helpers/parseFontface');
+
+// Pixel 5 XL
+const DEFAULT_VIEWPORT = {
+  width: 393,
+  height: 851,
+};
+
+/**
+ * Renders a page in Puppeteer and collects all data required for the page experience recommendations.
+ */
+class PageAnalyzer {
+  /**
+   * @param config optional configuration
+   * @param config.debug enable debug output, default false
+   * @param config.viewport the viewport size, default Pixel 5XL
+   */
+  constructor(config = {}) {
+    this.viewport = config.viewport || DEFAULT_VIEWPORT;
+    this.debug = config.debug || false;
+    this.started = false;
+  }
+
+  /**
+   * Start puppeteer
+   */
+  async start() {
+    this.browser = await puppeteer.launch();
+    this.started = true;
+  }
+
+  /**
+   * Load a page in Puppeteer and collect data required by checks. Needs to be called after 'start'.
+   *
+   * @param {string} url the URL to analyze
+   * @return {Object} all data collected on the page
+   */
+  async execute(url) {
+    if (!this.started) {
+      throw new Error('Puppeteer not running, please call `start` first.');
```

Comment:
is there any reason you don't just call start?

Reply:
You're right, but somehow I preferred the more explicit symmetry of calling start and shutdown. It's an internal API though so I don't really care...